app/library/Auth/Auth.php

```diff
@@ -300,4 +303,40 @@ public function getUser()
 
         return false;
     }
+    
+    /**
+     * Returns the current token user
+     *
+     * @param string $token
+     * @return boolean
+     */
+    public function findFirstByToken($token)
+    {
+        $userToken = RememberTokens::findFirst([
+            'conditions' => 'token = :token:',
+            'bind'       => [
+                'token' => $token,
+            ],
+        ]);
+        
+        $this->user_id = ($userToken) ? $userToken->usersId : false; 
+
+        return $this->user_id;
+    }
+
+    /**
+     * Delete the current user token in session
+     */
+    public function deleteToken($userId) {
+        $user = RememberTokens::find([
+            'conditions' => 'usersId = :userId:',
+            'bind'       => [
+                'userId' => $userId
+            ]
+        ]);
+
+        if ($user) {
+            $user->delete();
+        }
```

Comment:
@alejosv Shouldn't we unset `$this->user_id` here?  Also could you please follow common code style and use
```php
public function deleteToken($userId)
{
```
instead of
```php
public function deleteToken($userId) {
```

Reply:
Sorry, my intention is remove completely global variables, since, aren't necessary. I'll modified for not used `$this->user_id`, and will use `$user_id`. the code would look like this:

```    
    ... 
    $user_id = ($userToken) ? $userToken->usersId : false; 
    return $user_id;
    ...
```
Could it be like that?

I'll correct the code style, was a typo.

Reply:
LGTM